sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3567,12 +3567,27 @@ object SQLConf {
 
   val SHUFFLE_DEPENDENCY_FILE_CLEANUP_ENABLED =
     buildConf("spark.sql.shuffleDependency.fileCleanup.enabled")
-      .doc("When enabled, shuffle files will be cleaned up at the end of Spark Connect " +
-        "SQL executions.")
+      .doc("(Deprecated since Spark 4.1, please set" +
+        " 'spark.sql.connect.shuffleDependency.fileCleanup.enabled'.)")
       .version("4.0.0")
       .booleanConf
       .createWithDefault(Utils.isTesting)
 
+  val CLASSIC_SHUFFLE_DEPENDENCY_FILE_CLEANUP_ENABLED =
+    buildConf("spark.sql.classic.shuffleDependency.fileCleanup.enabled")
```

Comment:
@karuppayya shall we also add configs for thriftserver? I think it's another entry point that we can safely enable file cleanup.

Reply:
+1 , thanks for bringing this up
I will have this handled in a follow up PR soon